certbot/certbot/_internal/main.py

```diff
@@ -726,10 +727,28 @@ def _tos_cb(terms_of_service: str) -> None:
                 display_util.notify("Account registered.")
             except errors.MissingCommandlineFlag:
                 raise
-            except errors.Error:
+            except (errors.Error, acme_messages.Error) as err:
                 logger.debug("", exc_info=True)
+                if isinstance(err, acme_messages.Error):
+                    # `acme.messages.Error` has more details than `certbot.errors.Error`, so we
+                    # can use these details to provide a human readable error message:
+                    if err.detail:
+                        # `detail` contains a human readable error message provided by the
+                        # ACME server.
+                        err_msg = err.detail
+                    elif err.description:
+                        # If `detail` is not provided, the `acme` library can return a human
+                        # readable error message based on the error type in `description`.
+                        err_msg = err.description
+                    else:
+                        # If no human readable error message can be found, the ACME error type
+                        # is presented to the user.
+                        err_msg = str(err)
+                else:
+                    err_msg = str(err)
```

Comment:
Sorry for the delay reviewing this, it's been hard to find time.

For consistency and simplicity's sake, ISTM that we can just take advantage of `__str__` on `acme.messages.Error`:

https://github.com/certbot/certbot/blob/142fcad28b0e61f335b92a743b38c8d677bd00dd/acme/acme/messages.py#L137-L141

Would you be able to give that a try and see whether `str(err_msg)` does the trick? I'd like to avoid having one-off code related to display here. If we do end up needing a new way to display errors to users, we should deal with it in `acme`.

Reply:
I noticed that function too when looking at it. But if we use `str(err)` it would mean a Certbot raised error would be a relative simple error message which would be *very* different than an error message raised by the ACME client, as that would be a long line with all the different fields separated by " :: ". To me, that difference was too big, so I build the code above to make it more the same with regard to a user experience: a simple to understand error message.

But of course [KISS](https://en.wikipedia.org/wiki/KISS_principle) is always a good argument :wink:

Reply:
That's reasonable, and there's always the option to change what `__str__` does, if we do decide it's unreadable.

https://datatracker.ietf.org/doc/html/rfc7807#section-3.1 is pretty clear that only `title` and `detail` are supposed to be human-readable.

I also recall reading a comment on the Certbot repo years ago from one of Boulder's developers, calling Certbot out for the way the `description` property is interpreting the problem `type` and mapping it into [this dictionary](https://github.com/certbot/certbot/blob/142fcad28b0e61f335b92a743b38c8d677bd00dd/acme/acme/messages.py#L35-L66). This was a misinterpretation of the ACME document, I think.

tl;dr; I'm sure we'd be open to changing it, but let's kick it to another issue.

Reply:
I'll make this code as simple as possible, so we can change anything, if required, from a single point in `__str__()` in the future.